src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -240,11 +248,13 @@ private String getPasswordValue() {
 	}
 
 	static final class Build implements TypeSpec, AddressSpec, Builder {
+
 		String username;
 		Function<? super String, ? extends String> password;
 		String host;
 		int port;
 		Supplier<? extends InetSocketAddress> address;
+		Predicate<SocketAddress> nonProxyHostPredicate = RegexShouldProxyPredicate.ALWAYS_PROXY;
```

Comment:
From performance point of view which performs better `a -> false` or `pattern.matcher(isa.getHostString()).matches()`

Reply:
I was going to leave the lambda on ```Builder``` as a constant but then declaring something like
```
private static final Predicate<SocketAddress> ALWAYS_PROXY = a -> false
```
caused build warnings

```
warning: [UnnecessaryLambda] Returning a lambda from a helper method or saving it in a constant is unnecessary; prefer to implement the functional interface method directly and use a method reference instead.
```

instead of adding a static method on the builder or making it non-constant, I chose to create a trivial implementation ```AlwaysProxyPredicate``` but I am happy to move this.

Reply:
Yes this warning is so that you think about whether you want really to use the lambda as a constant. It depends on your use case. 
You can suppress it here, like the other one in this class `NO_HTTP_HEADERS`

So let's do it like this

```
@SuppressWarnings("UnnecessaryLambda")
private static final Predicate<SocketAddress> ALWAYS_PROXY = a -> false
```